src/inline-assembly.md

```diff
@@ -477,3 +478,135 @@ To avoid undefined behavior, these rules must be followed when using function-sc
   - The compiler is currently unable to detect this due to the way inline assembly is compiled, but may catch and reject this in the future.
 
 > **Note**: As a general rule, the flags covered by `preserves_flags` are those which are *not* preserved when performing a function call.
+
+### Directives Support
+
+Inline ASM supports a subset of the directives supported by both GNU AS and LLVM's internal assembler, given as follows.
+The result of using other directives is assembler-specific (and may cause an error, or may be accepted as-is).
+
+The following directives are guaranteed to be supported by the assembler:
+
+
+- `.2byte`
+- `.4byte`
+- `.8byte`
+- `.byte`
+- `.short`
+- `.word`
+- `.long`
+- `.quad`
+- `.float`
+- `.double`
+- `.octa`
+- `.sleb128`
+- `.uleb128`
+- `.ascii`
+- `.asciz`
+- `.string`
+- `.skip`
+- `.space`
+- `.balign`
+- `.balignl`
+- `.balignw`
+- `.balign`
+- `.balignl`
+- `.balignw`
+- `.fill`
+- `.section`
+- `.pushsection`
+- `.popsection`
+- `.text`
+- `.bss`
+- `.data`
+- `.def`
+- `.endef`
+- `.scl`
+- `.comm`
+- `.lcomm`
+- `.option`
+- `.equ`
+- `.set`
+- `.align`
+- `.inst`
+- `.p2align`
+
+
+The following directives are guaranteed to be supported for `global_asm` only:
+
+- `.alt_entry`
+- `.private_extern`
+- `.globl`
+- `.global`
+- `.size`
+- `.type`
+
+
+#### Target Specific Directive Support
+
+##### Dwarf Unwinding
+
+The following directives are supported on ELF targets that support DWARF unwind info:
+
+
+- `.cfi_adjust_cfa_offset`
+- `.cfi_def_cfa`
+- `.cfi_def_cfa_offset`
+- `.cfi_def_cfa_register`
+- `.cfi_endproc`
+- `.cfi_escape`
+- `.cfi_lsda`
+- `.cfi_offset`
+- `.cfi_personality`
+- `.cfi_register`
+- `.cfi_rel_offset`
+- `.cfi_remember_state`
+- `.cfi_restore`
+- `.cfi_restore_state`
+- `.cfi_return_column`
+- `.cfi_same_value`
+- `.cfi_sections`
+- `.cfi_signal_frame`
+- `.cfi_startproc`
+- `.cfi_undefined`
+- `.cfi_window_save`
+
+
+##### Structured Exception Handling
+
+On targets with structured exception Handling, the following additional directives are guaranteed to be supported:
+
+- `.seh_endproc`
+- `.seh_endprologue`
+- `.seh_proc`
+- `.seh_pushreg`
+- `.seh_savereg`
+- `.seh_setframe`
+- `.seh_stackalloc`
+
+
+##### x86
+
+On x86, the following additional directives are guaranteed to be supported:
+- `.nops`
+
+On x86 for `global_asm!` only, the following additional directives are guaranteed to be supported (it is unspecified whether `.code16` or `.code32` are supported for `asm!()`):
+
+- `.code16`
+- `.code32`
+
```

Comment:
```suggestion

The directives `.att_syntax` and `.intel_syntax` (including options such as `noprefix`) are not fully supported. Using these directives to affect literal assembly syntax with no operands will work, as long as the state is restored before the end of the assembly block. Using these directives to affect assembly that includes operands is not defined.

```

Reply:
You already have to support both syntaxes, given the att_syntax option.

I agree that the state restoration is covered above, but I felt that it couldn't hurt to remind people that this in particular is something stateful.